src/axom/primal/operators/detail/clip_impl.hpp

```diff
@@ -400,85 +400,95 @@ AXOM_HOST_DEVICE void poly_clip_reindex(Polyhedron<T, NDIMS>& poly,
 }
 
 /*!
- * \brief Finds the clipped intersection Polyhedron between Polyhedron
- *        poly and an array of Planes.
- *
- * \param [in] poly The polyhedron
- * \param [in] planes The array of planes
- * \param [in] eps The tolerance for plane point orientation.
+ * \brief Finds the polyhedron formed from clipping a polyhedron with a plane
  *
- * \return The Polyhedron formed from clipping the polyhedron with a set of planes.
+ * \param [in] poly The polyhedron to clip
```

Comment:
It looks like `poly` is an `[inout]` parameter since the implementation modifies it.
Please update the doxycomment.

Could this function instead return a bool indicating if the intersection is non-empty, and use the `[inout] poly` parameter without the extra copy for the return type?

If `clipPolyhedron` is used elsewhere, we'll have to modify those calls as well. The good news is that its an internal/detail function, so it's not meant to be used outside our internal implementation.

Reply:
I updated the doxygen to reflect it is an inout parameter and changed the return type to void. I experimented with returning a bool, but the following code pattern seems to say the clip failed rather than returning an empty polyhedron so I decided against it.

`if (!clipPolyhedron(poly, plane, eps))`

Reply:
Another option would be to return a reference to the original polyhedron to enable chaining.

`if (clipPolyhedron(poly, plane, eps).numVertices() == 0)`
`return clipPolyhedron(poly, plane, eps);`

Reply:
That's an interesting idea!

Reply:
I think I'll leave the return type as void for now. But if anyone requests chaining in the future, I'd be happy to implement it.